src/test/ui/consts/const-fn-extern.rs

```diff
@@ -0,0 +1,15 @@
+// run-pass
```

Comment:
Can you put the test in a directory `const-extern-fn` under `consts/` (also move the feature gate test there).

I'd also like to see more semantic tests:

1. `const unsafe extern fn` requires an `unsafe` context to be callable and errors because it was not in such a context.

2. Test that the `min_const_fn` restrictions apply, e.g. floating point arithmetic is not allowed, ptr-to-int casts are unstable, etc. (Consult similar tests in `consts/min_const_fn/`.)

   - In particular we should also ensure that a regular `extern fn` is not callable in a `const extern fn` either directly or through a function pointer. This should also include those defined in an `extern` block item.

3. Tests around coercions to `extern fn(...)` pointer types.

(The purpose of these tests is to ensure that everything we expect holds and that there's no strange interactions.)

Reply:
> also move the feature gate test there

Aren't feature-gate tests supposed to be in the `feature-gates` directory?

Reply:
Does the rustc guide say that (if so it should be changed imo)? -- From a T-Lang perspective I think it's better to just have all the tests for a feature in one place.

Reply:
It does: https://rust-lang.github.io/rustc-guide/implementing_new_features.html

> 5. Add a test to ensure the feature cannot be used without a feature gate, by creating feature-gate-$feature_name.rs and feature-gate-$feature_name.stderr files under the src/test/ui/feature-gates directory.

I think discussion of the placement of feature-gate tests should happen seapartely from this PR.

Reply:
Those docs are just outdated -- it doesn't reflect current practices. Fixing them in https://github.com/rust-lang/rustc-guide/pull/455.